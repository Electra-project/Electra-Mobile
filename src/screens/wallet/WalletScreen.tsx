import ElectraJs, { CoinMarketCapCurrency } from 'electra-js'
import numeral from 'numeral'
import React from 'react'
import { StatusBar, StyleSheet, Text, View } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'

const styles = StyleSheet.create({
  background: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center'
  },
  balanceInEca: {
    color: 'rgba(255, 255, 255, 1)',
    fontSize: 36,
    fontWeight: '700'
  },
  balanceInFiat: {
    color: 'rgba(255, 255, 255, .8)',
    fontSize: 21,
    fontWeight: '300'
  },
  balanceLabel: {
    color: 'rgba(255, 255, 255, .6)',
    fontSize: 18,
    fontWeight: '500'
  },
  container: {
    flex: 1
  }
})

export interface State {
  currency: CoinMarketCapCurrency
  balance: number
  currentPrice: number
}

export default class WalletScreen extends React.Component<{}, State> {
  private readonly electraJs: ElectraJs

  public constructor(props: object) {
    super(props)

    this.electraJs = new ElectraJs()
    this.state = {
      balance: 0,
      currency: 'USD',
      currentPrice: 0
    }
  }

  public componentDidMount() {
    this.electraJs.webServices.getCurrentPriceIn(this.state.currency)
      .then((currentPrice: number) => this.setState({
        balance: 1122300.724543798,
        currentPrice
      }))
  }

  public render(): JSX.Element {
    return (
      <View style={styles.container}>
        <StatusBar
          backgroundColor='#590c76'
          barStyle='light-content'
        />
        <LinearGradient
          colors={['#9619b3', '#6f1294']}
          end={{ x: 1, y: 0 }}
          start={{ x: 0, y: 0 }}
          style={styles.background}
        >
          <Text style={styles.balanceLabel}>
            YOUR TOTAL BALANCE
          </Text>
          <Text style={styles.balanceInEca}>
            {this.state.balance === 0
              ? '...'
              : numeral(this.state.balance).format('0,0.000')
            } ECA
          </Text>
          <Text style={styles.balanceInFiat}>
            {this.state.currentPrice === 0
              ? '...'
              : numeral(this.state.balance * this.state.currentPrice).format('0,0.00')
            }
            {this.state.currency}
          </Text>
        </LinearGradient>
      </View>
    )
  }
}
