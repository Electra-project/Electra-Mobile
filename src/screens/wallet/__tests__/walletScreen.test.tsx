import React from 'react'
import { shallow } from 'enzyme'

import WalletScreen from '../WalletScreen'

describe('WalletScreen', () => {
    let wrapper
    beforeAll(() => {
        wrapper = shallow(<WalletScreen />)
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('sets the starting state correctly', () => {
        const { currency, balance, currentPrice } = wrapper.state()
        expect(currency).toBe('USD')
        expect(balance).toBe(0)
        expect(currentPrice).toBe(0)
    })
})
