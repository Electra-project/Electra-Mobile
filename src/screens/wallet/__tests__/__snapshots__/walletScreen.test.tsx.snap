// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletScreen renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WalletScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <StatusBar
          animated={false}
          backgroundColor="#590c76"
          barStyle="light-content"
          showHideTransition="fade"
        />,
        <LinearGradient
          colors={
            Array [
              "#9619b3",
              "#6f1294",
            ]
          }
          end={
            Object {
              "x": 1,
              "y": 0,
            }
          }
          start={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, .6)",
                "fontSize": 18,
                "fontWeight": "500",
              }
            }
          >
            YOUR TOTAL BALANCE
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 36,
                "fontWeight": "700",
              }
            }
          >
            ...
             ECA
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "rgba(255, 255, 255, .8)",
                "fontSize": 21,
                "fontWeight": "300",
              }
            }
          >
            ...
            USD
          </Text>
        </LinearGradient>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animated": false,
          "backgroundColor": "#590c76",
          "barStyle": "light-content",
          "showHideTransition": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(255, 255, 255, .6)",
                  "fontSize": 18,
                  "fontWeight": "500",
                }
              }
            >
              YOUR TOTAL BALANCE
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "fontSize": 36,
                  "fontWeight": "700",
                }
              }
            >
              ...
               ECA
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(255, 255, 255, .8)",
                  "fontSize": 21,
                  "fontWeight": "300",
                }
              }
            >
              ...
              USD
            </Text>,
          ],
          "colors": Array [
            "#9619b3",
            "#6f1294",
          ],
          "end": Object {
            "x": 1,
            "y": 0,
          },
          "start": Object {
            "x": 0,
            "y": 0,
          },
          "style": Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "YOUR TOTAL BALANCE",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "rgba(255, 255, 255, .6)",
                "fontSize": 18,
                "fontWeight": "500",
              },
            },
            "ref": null,
            "rendered": "YOUR TOTAL BALANCE",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "...",
                " ECA",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 36,
                "fontWeight": "700",
              },
            },
            "ref": null,
            "rendered": Array [
              "...",
              " ECA",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "...",
                "USD",
              ],
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "rgba(255, 255, 255, .8)",
                "fontSize": 21,
                "fontWeight": "300",
              },
            },
            "ref": null,
            "rendered": Array [
              "...",
              "USD",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <StatusBar
            animated={false}
            backgroundColor="#590c76"
            barStyle="light-content"
            showHideTransition="fade"
          />,
          <LinearGradient
            colors={
              Array [
                "#9619b3",
                "#6f1294",
              ]
            }
            end={
              Object {
                "x": 1,
                "y": 0,
              }
            }
            start={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(255, 255, 255, .6)",
                  "fontSize": 18,
                  "fontWeight": "500",
                }
              }
            >
              YOUR TOTAL BALANCE
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "fontSize": 36,
                  "fontWeight": "700",
                }
              }
            >
              ...
               ECA
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "rgba(255, 255, 255, .8)",
                  "fontSize": 21,
                  "fontWeight": "300",
                }
              }
            >
              ...
              USD
            </Text>
          </LinearGradient>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animated": false,
            "backgroundColor": "#590c76",
            "barStyle": "light-content",
            "showHideTransition": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgba(255, 255, 255, .6)",
                    "fontSize": 18,
                    "fontWeight": "500",
                  }
                }
              >
                YOUR TOTAL BALANCE
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgba(255, 255, 255, 1)",
                    "fontSize": 36,
                    "fontWeight": "700",
                  }
                }
              >
                ...
                 ECA
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "rgba(255, 255, 255, .8)",
                    "fontSize": 21,
                    "fontWeight": "300",
                  }
                }
              >
                ...
                USD
              </Text>,
            ],
            "colors": Array [
              "#9619b3",
              "#6f1294",
            ],
            "end": Object {
              "x": 1,
              "y": 0,
            },
            "start": Object {
              "x": 0,
              "y": 0,
            },
            "style": Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "YOUR TOTAL BALANCE",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "rgba(255, 255, 255, .6)",
                  "fontSize": 18,
                  "fontWeight": "500",
                },
              },
              "ref": null,
              "rendered": "YOUR TOTAL BALANCE",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "...",
                  " ECA",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "fontSize": 36,
                  "fontWeight": "700",
                },
              },
              "ref": null,
              "rendered": Array [
                "...",
                " ECA",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "...",
                  "USD",
                ],
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "rgba(255, 255, 255, .8)",
                  "fontSize": 21,
                  "fontWeight": "300",
                },
              },
              "ref": null,
              "rendered": Array [
                "...",
                "USD",
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
